mitodl/micromasters
# util/util.js cleanup

#### What are the relevant tickets?

closes #438 

#### What's this PR do?

This PR basically does two cleanup tasks:

- move validation-related functions out of `util/util.js` and into `util/validation.js`, and move associated tests from `util/util_test.js` to `util/validation_test.js`
- move course list display related functions out of `util/util.js` and into `util/courseList.js`, ditto w/r/t tests

#### Where should the reviewer start?

There's a lot of changes, but it's really just moving code around.

#### How should this be manually tested?

Make sure there are no regressions in functionality, and that I didn't miss an import or a rename somewhere. Also, make sure no tests got lost in the shuffle (although I checked and there are the same number of tests run on master and on this branch).

#### Any background context you want to provide?

`util/util.js` was just getting really big, and wasn't really confined to 'utility' functions (general things) but was being used for render helpers that are not really general utility functions and the validation functions, which deserve to be in the same file as their compatriots.

#### Screenshots (if appropriate)

#### What GIF best describes this PR or how it makes you feel?
